web/pdf_find_controller.js

```diff
@@ -41,447 +42,440 @@ var CHARACTERS_TO_NORMALIZE = {
-        }
-      }
-      // Matches were not found (and searching is not done).
-      return false;
-    },
-
-    /**
-     * The method is called back from the text layer when match presentation
-     * is updated.
-     * @param {number} pageIndex - page index.
-     * @param {number} index - match index.
-     * @param {Array} elements - text layer div elements array.
-     * @param {number} beginIdx - start index of the div array for the match.
-     */
-    updateMatchPosition: function PDFFindController_updateMatchPosition(
-        pageIndex, index, elements, beginIdx) {
-      if (this.selected.matchIdx === index &&
-          this.selected.pageIdx === pageIndex) {
-        var spot = {
-          top: FIND_SCROLL_OFFSET_TOP,
-          left: FIND_SCROLL_OFFSET_LEFT,
-        };
-        scrollIntoView(elements[beginIdx], spot,
-                       /* skipOverflowHiddenElements = */ true);
+    }
+    // Start searching through the page.
+    this.nextPageMatch();
+  }
+
+  matchesReady(matches) {
+    let offset = this.offset;
+    let numMatches = matches.length;
+    let previous = this.state.findPrevious;
+
+    if (numMatches) {
+      // There were matches for the page, so initialize `matchIdx`.
+      this.hadMatch = true;
+      offset.matchIdx = (previous ? numMatches - 1 : 0);
+      this.updateMatch(true);
+      return true;
+    }
+    // No matches, so attempt to search the next page.
+    this.advanceOffsetPage(previous);
+    if (offset.wrapped) {
+      offset.matchIdx = null;
+      if (this.pagesToSearch < 0) {
+        // No point in wrapping again, there were no matches.
+        this.updateMatch(false);
+        // While matches were not found, searching for a page
+        // with matches should nevertheless halt.
+        return true;
       }
-    },
+    }
+    // Matches were not found (and searching is not done).
+    return false;
+  }
+
+  /**
+   * Called from the text layer when match presentation is updated.
+   *
+   * @param {number} pageIndex - The index of the page.
+   * @param {number} matchIndex - The index of the match.
+   * @param {Array} elements - Text layer `div` elements.
+   * @param {number} beginIdx - Start index of the `div` array for the match.
+   */
+  updateMatchPosition(pageIndex, matchIndex, elements, beginIdx) {
+    if (this.selected.matchIdx === matchIndex &&
+        this.selected.pageIdx === pageIndex) {
+      let spot = {
+        top: FIND_SCROLL_OFFSET_TOP,
+        left: FIND_SCROLL_OFFSET_LEFT,
+      };
+      scrollIntoView(elements[beginIdx], spot,
+                     /* skipOverflowHiddenElements = */ true);
+    }
+  }
 
-    nextPageMatch: function PDFFindController_nextPageMatch() {
-      if (this.resumePageIdx !== null) {
-        console.error('There can only be one pending page.');
+  nextPageMatch() {
+    if (this.resumePageIdx !== null) {
+      console.error('There can only be one pending page.');
+    }
+
+    let pageIdx = null, matches = null;
```

Comment:
Unless I'm mistaken, you *only* need to define `matches` here since it's used outside of the loop. However, `let pageIdx = ...` can go inside of the loop since that is limited to its scope.